src/core/binding.c

```diff
@@ -1506,10 +1508,13 @@ QuicBindingDeliverDatagrams(
         BOOLEAN DropPacket = FALSE;
         if (QuicBindingShouldRetryConnection(
                 Binding, Packet, TokenLength, Token, &DropPacket)) {
-            return
-                QuicBindingQueueStatelessOperation(
-                    Binding, QUIC_OPER_TYPE_RETRY, DatagramChain);
-
+            if (IsInitial) {
+                return
+                    QuicBindingQueueStatelessOperation(
+                        Binding, QUIC_OPER_TYPE_RETRY, DatagramChain);
+            }
+            QuicPacketLogDrop(Binding, Packet, "Reject non-initial packet during retry.");
```

Comment:
```suggestion
            QuicPacketLogDrop(Binding, Packet, "Non-initial packet during retry.");
```
Changed my mind on the text.